src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -477,11 +477,17 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
             this.RaiseCompilationTaskStart(null, "OverallCompilation");
 
             // loading the content to compiler
-            BondSchemas.Protocols.Initialize();
             this.logger = logger;
             this.LoadDiagnostics = ImmutableArray<Diagnostic>.Empty;
             this.config = options ?? default;
 
+            // When loading references is done through the generated C# and the syntax tree is not serialized,
+            // Bond protocols are not needed.
+            if (!this.config.LoadReferencesBasedOnGeneratedCsharp && !this.config.SerializeSyntaxTree)

```

Comment:
Based on the comment, is the logic supposed to be like this?
```suggestion
            if (!this.config.LoadReferencesBasedOnGeneratedCsharp || this.config.SerializeSyntaxTree)
```

Reply:
Since `LoadReferencesBasedOnGeneratedCsharp` is directly related to the deserializer and `SerializeSyntaxTree` is directly related to the serializer, I added and used new Bond protocols APIs to be able to initialize each one independently. This is an improvement over the previous setup since the compiler only initializes what is strictly needed.